### DISCUSSION TITLE
LG ThinQ AuthenticationError: Already in use by 3rd-Party Registeration:APPL

### DISCUSSION THREAD

**Discussion by Koksieboy** [nVuln/homebridge-lg-thinq#77] [Q&A]
Following the reinstallation and configuration of the LG ThinQ pluggin, I've been receiving the following errors in the log and the LG accessories are not showing.

ERRORUnknown Error:  AuthenticationError: Already in use by 3rd-Party Registeration:APPL
    at /var/lib/hoobs/lgthinq/node_modules/homebridge-lg-thinq/src/lib/Auth.ts:60:13
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
    at Auth.login (/var/lib/hoobs/lgthinq/node_modules/homebridge-lg-thinq/src/lib/Auth.ts:50:17)
    at API.ready (/var/lib/hoobs/lgthinq/node_modules/homebridge-lg-thinq/src/lib/API.ts:201:22)
    at ThinQ.isReady (/var/lib/hoobs/lgthinq/node_modules/homebridge-lg-thinq/src/lib/ThinQ.ts:236:7)
    at LGThinQHomebridgePlatform.discoverDevices (/var/lib/hoobs/lgthinq/node_modules/homebridge-lg-thinq/src/platform.ts:69:47)
    at HomebridgeAPI.<anonymous> (/var/lib/hoobs/lgthinq/node_modules/homebridge-lg-thinq/src/platform.ts:53:7)
11/09/2021, 01:01:12LG ThinQLGThinQThinQ API is not ready. please check configuration and try again.

Any help to get the pluggin working again would be greatly appreciated.

Thanks in advance.

**Comment by nVuln** [comment]
Is your LG account login type is logged by Apple ID ?
go to plugin setting page, use this feature to login

<img width="799" alt="Screen Shot 2021-09-11 at 11 15 18 AM" src="https://user-images.githubusercontent.com/54855446/132935957-917aafff-ef74-4af0-a367-207359b58cd8.png">

**Reply by Koksieboy** [reply]
Hi and thanks for your quick response. Yes, my login is using Apple authentication but I'm running Hoobs 4.0 and only have the options to login with LG account or token as shown below

![image](https://user-images.githubusercontent.com/90482983/132941883-3452603d-0b50-4d65-a168-4fa382a95e72.png)

**Reply by nVuln** [reply]
You can login by using this URL: [click here to login](https://us.m.lgaccount.com/spx/login/signIn?country=US&language=en-US&client_id=LGAO221A02&svc_list=SVC202&svc_integrated=Y&redirect_uri=https%3A%2F%2Fus.m.lgaccount.com%2Fspx%2Flogin%2FiabClose&show_thirdparty_login=LGE%2CMYLG%2CGGL%2CAMZ%2CFBK%2CAPPL&division=ha&callback_url=https%3A%2F%2Fus.m.lgaccount.com%2Fspx%2F)
when logged successfully, you may redirected to blank page (may be error page `page not found`), don't worry, just copy that URL and you can find `refresh_token` in this URL, it will look like `https://us.m.lgaccount.com/spx/login/iabClose?access_token=*******&refresh_token=***this is what you need***&oauth2_backend_url=https://us.lgeapi.com/`
then you put `refresh token` to plugin setting, change Auth Mode to `token` also

Another way, you can create new LG account (not logged by 3rd-party provider) and migrate all device to new account

**Reply by Koksieboy** [reply]
I've created a new account on LG and migrated all the devices to it. Everything is working fine now. Thanks a million for your help.

**Reply by jbh8063** [reply]
@nVuln I'm having a similar issue.  I'm using HOOBS 4, and my LG ThinQ app, using Google authentication, shows four devices; refrigerator, washer, dryer and dishwasher.

I tried using my LG credentials (same email address), and I get logged in, but no devices found:
4/8/2022, 4:50:18 PMLg Thinq Bridge starting
4/8/2022, 4:50:24 PMLg Thinq BridgeLoaded plugin 'homebridge-lg-thinq'
4/8/2022, 4:50:24 PMLg Thinq BridgeLoading 1 platforms...
4/8/2022, 4:50:24 PMLg Thinq BridgeBridge is running on port 51846.
4/8/2022, 4:50:28 PMLg Thinq BridgeLGThinQSuccessfully connected to the ThinQ API.
4/8/2022, 4:50:31 PMLg Thinq BridgeLGThinQWARNINGNo ThinQ devices in your account.

When I try to use the link above you posted 9/11/21 to get the refresh token, using the Google button, I'm getting a url like you pasted above starting w/ https://us.m.lgaccount.com  .... access_token .... token_type .... access_token .... but no REFRESH_TOKEN.  Trying that, as well as LG account using my Gmail credentials - both methods I get this:

4/8/2022, 5:35:11 PMLg Thinq BridgeLoaded plugin 'homebridge-lg-thinq'
4/8/2022, 5:35:11 PMLg Thinq BridgeLoading 1 platforms...
4/8/2022, 5:35:11 PMLg Thinq BridgeBridge is running on port 51846.
4/8/2022, 5:35:18 PMLg Thinq BridgeLGThinQERRORThinQ API is not ready. please check configuration and try again.
4/8/2022, 5:35:18 PMLg Thinq BridgeLGThinQERRORRequest failed with status code 406


 - John

**Reply by nVuln** [reply]
@jbh8063 at this time, logged by 3rd maybe got some issue, please following this wiki to grant access [Plugin Authorization Setting
](https://github.com/nVuln/homebridge-lg-thinq/wiki/Plugin-Authorization-Setting#if-your-account-logged-by-google-account-or-apple-id-or-facebook-account-using-third-party-login)
in your case, you should login to old account (logged by Google account), then send an invite access to your new account (LG credentials)

**Reply by jbh8063** [reply]
I was about to post that it still wasn't working after linking the second LG account, but then I found the json included the token info from my last attempt, even though I had LG Account selected.  I cleared the token info and it seems to have worked.  I do see this error in the log:

4/12/2022, 1:47:14 PMLg Thinq BridgeLGThinQAdding new accessory: e577e2a6-791b-1d36-8618-acf108809245: Refrigerator (REFRIGERATOR 2REF11EII__M2)
4/12/2022, 1:47:15 PMLg Thinq BridgeERROR[Refrigerator@Fridge@Target Temperature] characteristic was supplied illegal value: number 0.6 exceeded minimum of 1

But it appears to have added the refrigerator as an accessory in Hoobs, but it did not get pulled into Homekit like the Washer, Dryer and Dishwasher did.  Maybe that's not a big deal - I can't imagine setting up an automation on it

**Reply by nVuln** [reply]
@jbh8063 please turn on homebridge debug mode and see what happened, did you try enable thinq1 support in plugin setting ?

**Reply by jbh8063** [reply]
I'm not sure if I missed it the first time, or if somehow there was a delay, but I later found the refrigerator did, indeed, pull into HK.  Now what to do with it now that it's there, lol.  Thanks for your help, unless you want to dig into that error more; it doesn't appear to be a hard stop.
